src/services/embeddings_service.py

```
@@ -0,0 +1,278 @@
+        Returns:
+            List of embedding vectors
+        """
+        if not texts:
+            return []
+
+        batch_size = batch_size or self.config.batch_size
+        all_embeddings = []
+
+        # Process in batches
+        for i in range(0, len(texts), batch_size):
+            batch = texts[i : i + batch_size]
```

Flake8 (7.2.0)

[error] whitespace before ':'

(E203)